src/main/java/in/projecteka/consentmanager/consent/ConsentConfiguration.java

```diff
@@ -140,13 +156,22 @@ public ConsentRequestNotificationListener consentRequestNotificationListener(
     @SneakyThrows
     @Bean
     public KeyPair keyPair() {
-        KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
-        keyGen.initialize(2048);
-        return keyGen.generateKeyPair();
+        KeyStore keyStore = KeyStore.getInstance(keyStoreType);
+        File file = new File(keyStoreFilePath);
+        char[] pwdArray = keyStorePassword.toCharArray();
+        keyStore.load(new FileInputStream(file), pwdArray);
+        Certificate certificate = keyStore.getCertificate(keyStoreAlias);
```

Comment:
same key used for pin verification and consent artefact sigining?

Reply:
No. pin verification key is still getting generated dynamically. I will quickly replace it